test/k8sT/Services.go

```diff
@@ -2285,6 +2292,12 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 					It("Tests NodePort", func() {
 						testNodePort(true, false, helpers.ExistNodeWithoutCilium(), 0)
 					})
+					It("Test NodePort with netfilterCompatMode=true", func() {
```

Comment:
This will slow down the testing, because you are redeploying Cilium. Also, the next test depends on a previous configuration. Please move the test case to the bottom of the Context where test cases redeploy Cilium.

Another thing, the case for the direct routing is missing.

Reply:
makes sense, moved  down`netfilter-compat` test case. I have added test case with direct routing too.